apps/root/src/state/balances/actions.ts

```diff
@@ -65,10 +69,40 @@ export const fetchPricesForAllChains = createAppAsyncThunk<void, void>(
   }
 );
 
+const fetchTokenDetails = createAppAsyncThunk<
+  Token,
+  { tokenAddress: string; chainId: number; walletAddress: string; tokenList: TokenList }
+>(
+  'balances/fetchTokenDetails',
+  async ({ tokenAddress, chainId, walletAddress, tokenList }, { dispatch, extra: { web3Service } }) => {
+    if (tokenList[tokenAddress]) {
+      return tokenList[tokenAddress];
+    }
+    const tokenContract = await web3Service.contractService.getERC20TokenInstance(chainId, tokenAddress, walletAddress);
+    const [name, symbol, decimals] = await Promise.all([
+      tokenContract.name(),
+      tokenContract.symbol(),
+      tokenContract.decimals(),
+    ]);
+
+    const customToken = toToken({
+      address: tokenAddress,
+      name: capitalizeFirstLetter(name),
```

Comment:
no need to capitalize first letter, we should use the one provided on the contract